components/uniset/src/enum_props.rs

```diff
@@ -16,249 +18,304 @@ pub enum EnumeratedProperty {
-    Devanagari = 10,
-    DivesAkuru = 190,
-    Dogra = 178,
-    Duployan = 135,
-    EgyptianHieroglyphs = 71,
-    Elbasan = 136,
-    Elymaic = 185,
-    Ethiopic = 11,
-    Georgian = 12,
-    Glagolitic = 56,
-    Gothic = 13,
-    Grantha = 137,
-    Greek = 14,
-    Gujarati = 15,
-    GunjalaGondi = 179,
-    Gurmukhi = 16,
-    Han = 17,
-    Hangul = 18,
-    HanifiRohingya = 182,
-    Hanunoo = 43,
-    Hatran = 162,
-    Hebrew = 19,
-    Hiragana = 20,
-    ImperialAramaic = 116,
-    Inherited = 1,
-    InscriptionalPahlavi = 122,
-    InscriptionalParthian = 125,
-    Javanese = 78,
-    Kaithi = 120,
-    Kannada = 21,
-    Katakana = 22,
-    KayahLi = 79,
-    Kharoshthi = 57,
-    KhitanSmallScript = 191,
-    Khmer = 23,
-    Khojki = 157,
-    Khudawadi = 145,
-    Lao = 24,
-    Latin = 25,
-    Lepcha = 82,
-    Limbu = 48,
-    LinearA = 83,
-    LinearB = 49,
-    Lisu = 131,
-    Lycian = 107,
-    Lydian = 108,
-    Mahajani = 160,
-    Makasar = 180,
-    Malayalam = 26,
-    Mandaic = 84,
-    Manichaean = 121,
-    Marchen = 169,
-    MasaramGondi = 175,
-    Medefaidrin = 181,
-    MeeteiMayek = 115,
-    MendeKikakui = 140,
-    MeroiticCursive = 141,
-    MeroiticHieroglyphs = 86,
-    Miao = 92,
-    Modi = 163,
-    Mongolian = 27,
-    Mro = 149,
-    Multani = 164,
-    Myanmar = 28,
-    Nabataean = 143,
-    Nandinagari = 187,
-    NewTaiLue = 59,
-    Newa = 170,
-    Nko = 87,
-    Nushu = 150,
-    NyiakengPuachueHmong = 186,
-    Ogham = 29,
-    OlChiki = 109,
-    OldHungarian = 76,
-    OldItalic = 30,
-    OldNorthArabian = 142,
-    OldPermic = 89,
-    OldPersian = 61,
-    OldSogdian = 184,
-    OldSouthArabian = 133,
-    OldTurkic = 88,
-    OldUyghur = 194,
-    Oriya = 31,
-    Osage = 171,
-    Osmanya = 50,
-    PahawhHmong = 75,
-    Palmyrene = 144,
-    PauCinHau = 165,
-    PhagsPa = 90,
-    Phoenician = 91,
-    PsalterPahlavi = 123,
-    Rejang = 110,
-    Runic = 32,
-    Samaritan = 126,
-    Saurashtra = 111,
-    Sharada = 151,
-    Shavian = 51,
-    Siddham = 166,
-    SignWriting = 112,
-    Sinhala = 33,
-    Sogdian = 183,
-    SoraSompeng = 152,
-    Soyombo = 176,
-    Sundanese = 113,
-    SylotiNagri = 58,
-    Syriac = 34,
-    Tagalog = 42,
-    Tagbanwa = 45,
-    TaiLe = 52,
-    TaiTham = 106,
-    TaiViet = 127,
-    Takri = 153,
-    Tamil = 35,
-    Tangsa = 195,
-    Tangut = 154,
-    Telugu = 36,
-    Thaana = 37,
-    Thai = 38,
-    Tibetan = 39,
-    Tifinagh = 60,
-    Tirhuta = 158,
-    Toto = 196,
-    Ugaritic = 53,
-    Unknown = 103,
-    Vai = 99,
-    Vithkuqi = 197,
-    Wancho = 188,
-    WarangCiti = 146,
-    Yezidi = 192,
-    Yi = 41,
-    ZanabazarSquare = 177,
+#[derive(Copy, Clone, Debug, Eq, PartialEq)]
+#[repr(transparent)]
+pub struct Script(pub(crate) u16);
+
+#[allow(missing_docs)] // These constants don't need documentation.
+#[allow(non_upper_case_globals)]
+impl Script {
+    pub const Adlam: Script = Script(167);
```

Comment:
Nit (optional): I still don't see the use case behind defining all these constants.

Reply:
I use these constants in `icu4x_js_regexp` (and also in `get_script_val_set`). I think a) there's a clear case for defining the more common constants, b) there's no clear line for how we would decide which scripts aren't important enough to define, and c) the cost of defining a constant is roughly zero, so we should err on the side of generosity to the user.